cpp/arcticdb/column_store/column.hpp

```diff
@@ -670,7 +679,22 @@ class Column {
         typename = std::enable_if_t<std::is_invocable_r_v<void, functor, typename input_tdt::DataTypeTag::raw_type>>>
     static void for_each(const Column& input_column, functor&& f) {
         auto input_data = input_column.data();
-        std::for_each(input_data.cbegin<input_tdt>(), input_data.cend<input_tdt>(), std::forward<functor>(f));
+        std::for_each(input_data.cbegin<input_tdt, false>(), input_data.cend<input_tdt, false>(), std::forward<functor>(f));
+    }
+
+    template <
+            typename input_tdt,
+            typename functor,
+            typename = std::enable_if_t<std::is_invocable_r_v<void, functor, ColumnData::Enumeration<typename input_tdt::DataTypeTag::raw_type, true>>>>
```

Comment:
nit: We can use concepts now instead of enable if, it's a bit easier on the eyes